scripts/glean.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import List
+
+import cachecontrol
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class _Cache:
+    """
+    Simple cache manager so we can avoid refetching the same dependency data
+    over and over
+    """
+
+    def __init__(self):
+        self.sess = requests.Session()
+        self.sess.mount("http://", cachecontrol.CacheControlAdapter())
+        self.sess.mount("https://", cachecontrol.CacheControlAdapter())
+
+    def get(self, url: str):
+        return self.sess.get(url)
+
+    def get_json(self, url: str):
+        return self.get(url).json()
+
+
+_cache = _Cache()
+
+
+class GleanObject(object):
+    NAME_KEY = "name"
+    HISTORY_KEY = "history"
+
+
+class GleanMetric(GleanObject):
+    """
+    Represents an individual Glean metric, as defined by probe scraper
+    """
+
+    ALL_PINGS_KEYWORDS = ("all-pings", "all_pings")
+
+    def __init__(self, identifier: str, definition: dict, *, ping_names: List[str] = None):
```

Comment:
What does the standalone `*` do here?

Reply:
Nothing useful. :) I think this is a holdover from what I copied over from mozilla-schema-generator, I'll take it out.